src/StabilizerProcessor.cs

```diff
@@ -233,7 +240,7 @@ private void WriteToScratch(IQArray<Pauli> bases, IQArray<Qubit> qubits, Qubit a
                         H(qubit);
                         S(qubit);
                         ControlledX(new QArray<Qubit>(new[] { qubit }), aux);
-                        S(qubit);
+                        SAdjoint(qubit);
```

Comment:
This totally was the problem I think!

Reply:
It was one of the problems. When I fixed that one, I thought I had it. But a lot of tests kept failing.